sdk/src/manifest_assertion.rs

```diff
@@ -0,0 +1,183 @@
+    data: ManifestData,
+    /// There can be more than one assertion for any label
+    #[serde(skip_serializing_if = "Option::is_none")]
+    instance: Option<usize>,
+    /// The [ManifestAssertionKind] for this assertion (as stored in c2pa content)
+    #[serde(skip_serializing_if = "Option::is_none")]
+    kind: Option<ManifestAssertionKind>,
+}
+
+// fn manifest_data_serialize<S>(x: &ManifestData, s: S) -> std::result::Result<S::Ok, S::Error>
+// where
+//     S: Serializer,
+// {
+//     s.serialize_str("<omitted>")
+// }
+
+// fn manifest_data_deserialize<'de, D>(deserializer: D) -> std::result::Result<ManifestData, D::Error>
+// where
+//     D: Deserializer<'de>,
+// {
+//     let s: &str = Deserialize::deserialize(deserializer)?;
+//     serde_json::from_str(s).map_err(D::Erro)
+// }
+
+impl ManifestAssertion {
+    /// Create with label and value
+    pub fn new(label: String, data: Value) -> Self {
+        Self {
+            label,
+            data: ManifestData::Json(data),
+            instance: None,
+            kind: None,
+        }
+    }
+
+    /// Returns the la
+    pub fn label(&self) -> &str {
+        &self.label
+    }
+
+    pub fn label_with_instance(&self) -> String {
+        match self.instance {
+            Some(i) if i > 1 => format!("{}__{}", self.label, i),
+            _ => self.label.to_owned(),
+        }
+    }
+
+    pub fn value(&self) -> Result<&Value> {
+        match &self.data {
+            ManifestData::Json(d) => Ok(d),
+            ManifestData::Binary(_) => Err(Error::UnsupportedType),
+        }
+    }
+
+    pub fn binary(&self) -> Result<&[u8]> {
+        match &self.data {
+            ManifestData::Json(_) => Err(Error::UnsupportedType),
+            ManifestData::Binary(b) => Ok(b),
+        }
+    }
+
+    pub fn instance(&self) -> usize {
+        self.instance.unwrap_or(1)
+    }
+
+    pub fn kind(&self) -> &ManifestAssertionKind {
+        match self.kind.as_ref() {
+            Some(kind) => kind,
+            None => &ManifestAssertionKind::Cbor,
+        }
+    }
+
+    pub(crate) fn set_instance(mut self, instance: usize) -> Self {
+        self.instance = if instance > 1 { Some(instance) } else { None };
+        self
+    }
+
+    pub fn set_kind(mut self, kind: ManifestAssertionKind) -> Self {
+        self.kind = Some(kind);
+        self
+    }
+
+    pub fn from_labeled_assertion<S: Into<String>, T: Serialize>(
+        label: S,
+        data: &T,
+    ) -> Result<Self> {
+        Ok(Self::new(
+            label.into(),
+            serde_json::to_value(data).map_err(|_err| Error::AssertionEncoding)?,
+        ))
+    }
+
+    pub fn from_helper<T: Serialize + AssertionBase>(data: &T) -> Result<Self> {
+        Ok(Self::new(
+            data.label().to_owned(),
+            serde_json::to_value(data).map_err(|_err| Error::AssertionEncoding)?,
+        ))
+    }
+
+    pub fn to_helper<T: DeserializeOwned>(&self) -> Result<T> {
```

Comment:
Would you be able to add comments to these functions (especially these two)? `helper` isn't a particularly descriptive term and it would be useful to have more insight into what this means.

Reply:
You can also do this as a follow-up PR if necessary - I see that you started on the `label` function.

Reply:
Comments and code examples added, also reverted the names until we decided if we want helper or not.